fix: display metadata block even without title/subtitle

mcanouil opened the pull request:
This PR tweaks the HTML template to include the metadata partial as header if no title or subtitle are provided.

Fixes #5856

mcanouil committed:
fix: Quarto doesn't print authors and affiliations defined in the YAML if the title is missing

Fixes #5856

mcanouil committed:
fix: remove duplicated header tag

dragonstyle commented on src/resources/formats/html/pandoc/template.html:
```diff
@@ -36,6 +36,8 @@
 $title-block.html()$
 $elseif(subtitle)$
 $title-block.html()$
+$elseif(by-author)$
```
I am thinking that this approach would screen out cases where there is no title, no subtitle, and author normalization is disabled, but the same point could be made about including a date (e.g. user put only a date and it didn't appear).

I would suggest we should just always include `title-block` without condition - what do you think?

mcanouil replied:
Makes sense, I made a small change in that direction: cfa2917b120af88bca230298fa9cb5bcdc80913a

mcanouil replied:
I made the change 1704e9aec878dec9cf91241af83c18e195a9b6ae and did few manual testing on simple HTML document.
Seems good.

mcanouil committed:
fix: add if title before add it

mcanouil committed:
fix: always write title block

The title block is now always written, even if the title is empty or "metadata" is provided for the title block.

cderv reviewed (commented):
Just thinking that this could lead to possible breakage or at least changes in output for users that would not have a `title` nor a `subtitle`. This will lead to a new `header` HTML node in the document. 

We should just write this in the changelog so that readers are aware. If we care about sharing such changes when we know it happens 😅 

users impacted by this should be rare though though.

mcanouil commented:
> We should just write this in the changelog so that readers are aware. If we care about sharing such changes when we know it happens 😅

Yes, that was my intent.
I usually fill the changelog once all agreed on changes made by the PR.

cderv commented:
Perfect ! 

I am not sure what we usually include in changelog. Sometime only the we fix an issue and add a feature like 

* metadata block will now be included in output even when no `title` or `subtitle`

This does not specifically explain that new `<header>` block will be added and will modify HTML outputs.  That is why I wanted to add the thoughts. But maybe this is not important. 

> I usually fill the changelog once all agreed on changes made by the PR.

Yep you're right to do that. 👍 I was not implying it was missing ! 😉

dragonstyle commented:
Ahh sorry I merged without the changeling addition - can you still add that in a separate commit @mcanouil ?

mcanouil commented:
Only via a new PR.

dragonstyle commented:
I can make that change since it's my fault. Sorry about that

mcanouil commented:
Feel free to make a commit directly in main.
Or I can make a PR in few hours.
Up to you.

cderv commented:
> I usually fill the changelog once all agreed on changes made by the PR.

See, you must be really quick here ! 😆
